### ISSUE TITLE
Panic when passing a deploy to `routes`

### ISSUE THREAD

**Issue by grampelberg** [linkerd/linkerd2#1915]
### What is the issue?

Running `routes` on a deployment throws a panic:

```bash
$ bin/go-run cli routes -n emojivoto deploy/web
panic: runtime error: index out of range
```

It would be nice to do some input validation and explain that `routes` can only take services (instead of any resource type).

**Comment by grampelberg** [comment]
@siggy has fixed this on master right now. I'd just like to see a new error message:

```bash
Error: the routes command can only take services, please use a valid service resource
Usage:
  linkerd routes [flags] (SERVICE)
```

**Comment by adleong** [comment]
Fixed by #1921
